src/server.ts

```diff
@@ -9,6 +10,9 @@ const INTERFACE: string = process.env.INTERFACE || '0.0.0.0';
 const app = fastify();
 app.register(FastifyCors, {});
 
+// Add version header
+addCustomVersionHeader(app);
```

Comment:
I'd prefer that you add the hook in the `registerRoutes()` function because then in the unit test it would be just a matter of doing:

```
describe('Chaos Stream Proxy server', () => {
  let app;
  beforeEach(() => {
    app = fastify();
    registerRoutes(app);
  });

  it.each([HLS_PROXY_MASTER, HLS_PROXY_MEDIA, SEGMENTS_PROXY_SEGMENT])('route %p contains x-version header', (route) => {
    const response = await app.inject(route);
    expect(response.headers).objectContaining(...)
  });
});
```